wandb/sdk/data_types/utils.py

```diff
@@ -92,10 +94,28 @@ def val_to_json(
 
             items = _prune_max_seq(val)
 
-            for i, item in enumerate(items):
-                item.bind_to_run(
-                    run, key, namespace, id_=i, ignore_copy_err=ignore_copy_err
-                )
+            if _server_accepts_image_filenames():
+                for item in items:
+                    item.bind_to_run(
+                        run=run,
+                        key=key,
+                        step=namespace,
+                        ignore_copy_err=ignore_copy_err,
+                    )
+            else:
+                for i, item in enumerate(items):
+                    item.bind_to_run(
+                        run=run,
+                        key=key,
+                        step=namespace,
+                        id_=i,
+                        ignore_copy_err=ignore_copy_err,
+                    )
+                if run._attach_id and run._init_pid != os.getpid():
```

Comment:
Ah, I see, so yes it might and we issue a warn in that case.

Reply:
yea exactly!